tests/php/Unit/Internal/Domain/PhelFnNormalizerTest.php

```diff
@@ -0,0 +1,310 @@
+        $phelFnLoader->method('getNormalizedPhelFunctions')->willReturn([
+            'fn-name-' => $this->createMock(PersistentMapInterface::class),
+            'FN-NAME' => $this->createMock(PersistentMapInterface::class),
+        ]);
+
+        $normalizer = new PhelFnNormalizer($phelFnLoader);
+        $actual = $normalizer->getNormalizedGroupedFunctions();
+
+        $expected = [
+            'fn-name' => [
+                NormalizedPhelFunction::fromArray([
+                    'fnName' => 'fn-name-',
+                    'doc' => '',
+                    'fnSignature' => '',
+                    'desc' => '',
+                ]),
+                NormalizedPhelFunction::fromArray([
+                    'fnName' => 'FN-NAME',
+                    'doc' => '',
+                    'fnSignature' => '',
+                    'desc' => '',
+                ]),
+            ],
+        ];
+
+        self::assertEquals($expected, $actual);
+    }
+
+    public function test_skip_private_symbol(): void
+    {
+        $privateSymbol = $this->createMock(PersistentMapInterface::class);
+        // Mocking the `$meta[Keyword::create('private')]`
+        $privateSymbol->method('offsetExists')->willReturn(true);
+        $privateSymbol->method('offsetGet')->willReturn(true);
+
+        $phelFnLoader = $this->createMock(PhelFnLoaderInterface::class);
+        $phelFnLoader->method('getNormalizedPhelFunctions')->willReturn([
+            'privateSymbol' => $privateSymbol,
+        ]);
+
+        $normalizer = new PhelFnNormalizer($phelFnLoader);
+
+        self::assertEmpty($normalizer->getNormalizedGroupedFunctions());
+    }
+
+    public function test_symbol_without_doc(): void
+    {
+        $symbol = $this->createStub(PersistentMapInterface::class);
+        $symbol->method('offsetExists')->willReturn(true);
+        // false -> relates to `isPrivate`
+        // null  -> relates to `doc`
+        $symbol->method('offsetGet')->willReturnOnConsecutiveCalls(false, null);
+
+        $phelFnLoader = $this->createMock(PhelFnLoaderInterface::class);
+        $phelFnLoader->method('getNormalizedPhelFunctions')->willReturn([
+            '*compile-mode*' => $symbol,
+        ]);
+
+        $normalizer = new PhelFnNormalizer($phelFnLoader);
+        $actual = $normalizer->getNormalizedGroupedFunctions();
+
+        $expected = [
+            'compile-mode' => [
+                NormalizedPhelFunction::fromArray([
+                    'fnName' => '*compile-mode*',
+                    'doc' => '',
+                    'fnSignature' => '',
+                    'desc' => '',
+                ]),
+            ],
+        ];
+
+        self::assertEquals($expected, $actual);
+    }
+
+    public function test_symbol_with_doc_and_desc(): void
+    {
+        $symbol = $this->createStub(PersistentMapInterface::class);
+        $symbol->method('offsetExists')->willReturn(true);
+        // false -> relates to `isPrivate`
+        // '...' -> relates to `doc`
```

Comment:
```suggestion
        // 'Constant for (...)' -> relates to `doc`
```
Maybe it's a bit clearer this way? 👀

Reply:
Or even?:
```php
$symbol->method('offsetGet')->willReturnOnConsecutiveCalls(
    false,                                     // relates to 'isPrivate'
    'Constant for Not a Number (NAN) values.', // relates to 'doc'
);
```